src/speech_recognition/kaldi_gstreamer_app.py

```diff
@@ -8,27 +8,36 @@
 
 # System imports
 import os
+import subprocess
 
 # Gstreamer imports
 import gi
 gi.require_version('Gst', '1.0')
-from gi.repository import GObject, Gst
+from gi.repository import Gst
 
 # Speech recognition
 from .gstreamer_app import GstApp
-
+from .kaldi_grammar import Grammar
 
 class KaldiGstApp(GstApp):
     """Kaldi Gstreamer Application"""
-    def __init__(self, model_path, grammar):
+    def __init__(self, model_path, grammar, target, ispreemt_requested):
         """Initialize a KaldiGstApp object"""
         GstApp.__init__(self)
 
         self.type = 'Kaldi-Gst-App'
+        self.grammar = Grammar(model_path, grammar, target)
+
+        # Prepare grammar and decoding graphs
+        subp_status = subprocess.call(["mkdynamicgraph.bash",
```

Comment:
relative path, so this fails if working dir is different

Reply:
I don't think so. In the env-hooks script I add this script to path so it can be called from any working directory directly.

Reply:
Does `subprocess.call()` source the `.bashrc`? Then it is isafe.

Reply:
I wouldn't say that is sources `.bashrc` but it keeps the environment as it is.
During the tests I did not modify my path in `.bashrc` but directly sourced the env-hooks script in my environment and the function scripts were available to `subprocess.call()`

Reply:
Ok, that is also oke.